resolwe_bio/tools/tximport_summarize.R

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env Rscript
+library(argparse)
+library(GenomicFeatures)
+library(tximport)
+
+parser = ArgumentParser(description='Summarize TPM values to gene-level expressions.')
+parser$add_argument('infile', help="Salmon output file (transcript-level).")
+parser$add_argument('outfile', help='GTF file used for transcript-to-gene mapping.')
+parser$add_argument('gtf', help='GTF file used for transcript-to-gene mapping.')
+args = parser$parse_args(commandArgs(trailingOnly=TRUE))
+
+# Create transcript-to-gene mapping table
+db_from_gtf = makeTxDbFromGFF(args$gtf)
+k = keys(db_from_gtf, keytype = "TXNAME")
+tx2gene = select(db_from_gtf, k, "GENEID", "TXNAME")
+# Save transcript-to-gene mapping table to file
+write.table(tx2gene, file='tx2gene.txt', sep="\t", quote = FALSE, row.names = FALSE)
```

Comment:
The output file name of this table should be a command-line parameter. In the process, `tx2gene` file name should be derived from the name of the annotation file.